ddtrace/tracer/tracer.go

```diff
@@ -390,7 +401,7 @@ func (t *tracer) pushPayload(trace []*span) {
 	if t.payload.size() > payloadSizeLimit {
 		// getting large
 		select {
-		case t.flushChan <- nil:
+		case t.flushChan <- struct{}{}:
```

Comment:
This `flushChan` channel and this whole `select` statement makes no sense. We can simply call the `flush` function directly and just drop this channel. Our "new `forceFlush`" will continue to work perfectly. We should definitely do that at some point. There are way too many channels in this tracer (e.g. syncPush, flushChan) which are leftover dinosaurs 🙂

Reply:
I think that's an easy enough change it's worth trying in this PR.  👍 
Then we can just be done with the flush channel.

Reply:
Awesome!

Reply:
🙂 Let me know if I can help